See debian bug #521857, patch from Timothy Redaelli <timothy@redaelli.eu>

--- ettercap-0.7.3.orig/src/protocols/ec_tcp.c
+++ ettercap-0.7.3/src/protocols/ec_tcp.c
@@ -116,7 +116,7 @@
    tcp = (struct tcp_header *)DECODE_DATA;
    
    opt_start = (u_char *)(tcp + 1);
-   opt_end = (u_char *)((int)tcp + tcp->off * 4);
+   opt_end = (u_char *)(tcp + tcp->off * 4);
 
    DECODED_LEN = (u_int32)(tcp->off * 4);
